### PR TITLE
Bryanv/1671 tick overrides

### LINKED ISSUE #1671: allow users to specify explicit tick labels
`FixedTicker` allows users to specify fixed tick locations, but also need to allow users to specify explicit tick labels

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #1671
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
Simple test:
```
from bokeh.io import output_file, show
from bokeh.plotting import figure

p = figure(x_range=(0,10))
p.circle(x=[2,3,4], y=[6,7,8])
p.xaxis.ticker = [0, 5, 10]
p.xaxis.major_label_overrides = {0: "zero", 5: "five", 10: "ten"}

output_file("foo.html")

show(p)
```
screen shot 2017-05-02 at 00 02 06

File: bokeh/models/axes.py [hunk 1]
```diff
@@ -37,7 +37,29 @@ class Axis(GuideRenderer):
 
     ticker = Instance(Ticker, help="""
     A Ticker to use for computing locations of axis components.
-    """)
+
+    THe property may also be passed a sequence of floating point numbers as
```
Comment (canavandl) [inline bokeh/models/axes.py]: Weird capitalization in "THe"

**Comment by canavandl** [comment]
LGTM. It's nicer than the FixedTicker/FuncTickFormatter with a lookup dict nonsense I was going previously.

**Comment by StevenCHowell** [comment]
This works great!  

For when this eventually gets into the documentation, I would recommend saving the dictionary and using the dictionary keys property for the `ticker`, e.g.,
```
from bokeh.io import output_file, show
from bokeh.plotting import figure

p = figure(x_range=(0,10))
p.circle(x=[2,3,4], y=[6,7,8])
label_dict = {0: "zero", 5: "five", 10: "ten"}
p.xaxis.ticker = label_dict.keys()
p.xaxis.major_label_overrides = label_dict 

output_file("foo.html")

show(p)
```

**Pull request merged**